### PR TITLE
feat: add Draft Unique validation for FormR PartB

### PR DESCRIPTION
**Pull request by hihilary**
Pretty much the same as FormRPartA. Added unique validation for FormR PartB draft.

### CODE REVIEW

File: src/test/java/uk/nhs/hee/tis/trainee/forms/api/validation/FormRPartBValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    formRPartB.setId("ANOTHER_ID");
+    formRPartB.setLifecycleState(LifecycleState.DRAFT);
+
+    when(formRPartBRepositoryMock
+        .findByTraineeTisIdAndLifecycleState(formRPartBDto.getTraineeTisId(),
+            formRPartBDto.getLifecycleState()))
+        .thenReturn(Lists.list(formRPartB));
+
+    List<FieldError> fieldErrors = validator.checkIfDraftUnique(formRPartBDto);
+    assertThat("Should not return any errors", fieldErrors.size(), is(1));
```
Comment (JosephKelly) [inline src/test/java/uk/nhs/hee/tis/trainee/forms/api/validation/FormRPartBValidatorTest.java:155]: I think this message is the only thing to change

Comment (hihilary) [reply src/test/java/uk/nhs/hee/tis/trainee/forms/api/validation/FormRPartBValidatorTest.java]: Sorry, my bad, again. Thought I missed one last time. Fixed them both.

Comment (JosephKelly) [reply src/test/java/uk/nhs/hee/tis/trainee/forms/api/validation/FormRPartBValidatorTest.java]: Ooops.  I missed the one in PartA too!

File: src/test/java/uk/nhs/hee/tis/trainee/forms/api/validation/FormRPartAValidatorTest.java [hunk 2]
```diff
@@ -152,7 +152,7 @@ void validateCreateDraftIfOneDraftFound() {
         .thenReturn(Lists.list(formRPartA));
 
     List<FieldError> fieldErrors = validator.checkIfDraftUnique(formRPartADto);
-    assertThat("Should not return any errors", fieldErrors.size(), is(1));
+    assertThat("Should not return 1 error", fieldErrors.size(), is(1));
```
Comment (JosephKelly) [inline src/test/java/uk/nhs/hee/tis/trainee/forms/api/validation/FormRPartAValidatorTest.java:155]: extra `not`

Comment (hihilary) [reply src/test/java/uk/nhs/hee/tis/trainee/forms/api/validation/FormRPartAValidatorTest.java]: Fixed. I was too stupid to miss it...

**Pull request merged**